WINC-726: Add VM hardware version in Packer config (#895)
openshift/windows-machine-config-operator: docs/vsphere-golden-image.md

> ## 2. Create the virtual machine

Reviewer: Remove `WINC-726`  from the commit message.

To start with, create a new virtual machine in vSphere from the selected Windows Server distribution using the ISO image.
See [vSphere documentation](https://docs.vmware.com/en/VMware-vSphere/index.html) for installation guidelines. We 

> recommend using a [virtual hardware version](https://kb.vmware.com/s/article/1003746) of `15` when creating the VM.

Reviewer: Consider moving this change to another commit with a clear commit message. 
For example:  
```
[docs] Set hardware version in vSphere

...
```

Easier to review by docs Team.

Author: I don't see this as necessary

Reviewer: See https://github.com/openshift/windows-machine-config-operator/pull/515/commits for references.

> See [vSphere documentation](https://docs.vmware.com/en/VMware-vSphere/index.html) for installation guidelines. We 
> recommend using a [virtual hardware version](https://kb.vmware.com/s/article/1003746) of `15` when creating the VM.

Reviewer: ```suggestion
See [vSphere documentation](https://docs.vmware.com/en/VMware-vSphere/index.html) for installation guidelines. The supported [virtual hardware version](https://kb.vmware.com/s/article/1003746) is `15`.
```

Reviewer: Re-reading this, we don't need to add another maintenance point WRT the hardware version.

For client-facing docs we should NOT set any explicit version, just a reference to [VMware vSphere infrastructure requirements](https://docs.openshift.com/container-platform/4.9/installing/installing_vsphere/preparing-to-install-on-vsphere.html#installation-vsphere-infrastructure_preparing-to-install-on-vsphere) for OCP will suffice, it reads:

To start with, create a new virtual machine in vSphere from the selected Windows Server distribution using the ISO image.
See [vSphere documentation](https://docs.vmware.com/en/VMware-vSphere/index.html) for installation guidelines.

> The supported [virtual hardware version](https://kb.vmware.com/s/article/1003746) is `15`.

Reviewer 2: My info might be out of date, but I thought we were going to link to the [OpenShift docs](https://docs.openshift.com/container-platform/4.9/updating/updating-hardware-on-nodes-running-on-vsphere.html) for this information?

This should be OpenShift release specific and 4.9 supports 13+?

Author: The 4.9 docs have a note:
```
virtual hardware version 13 is now deprecated. These versions are still fully supported, but support will be removed in a future version of OpenShift Container Platform. Hardware version 15 is now the default
```
But yes, I agree, I'll change this point to OpenShift docs and not mention a any specific version here.

Reviewer 2: Wow thats confusing as it contradicts the first paragraph of that page :laughing:
